pkg/controller-gen/generators/type_go.go

```diff
@@ -200,14 +205,38 @@ func (a *{{.lowerName}}GeneratingHandler) Handle(obj *{{.version}}.{{.type}}, st
 	}
 
 	objs, newStatus, err := a.{{.type}}GeneratingHandler(obj, status)
-	if err != nil {
+	if err != nil || !a.isNewResourceVersion(obj) {
 		return newStatus, err
 	}
```

Comment:
Similar to my other comment, this is logically the same it better describes our intent since we are maintaining templated code.
```suggestion
	if err != nil {
		return newStatus, err
	}
	if !a.isNewResourceVersion(obj) {
	    return newStatus, nil
	}
```